pycheribuild/boot_cheribsd/__init__.py

```diff
@@ -815,15 +823,23 @@ def boot_and_login(child: CheriBSDSpawnMixin, *, starttime, kernel_init_only=Fal
         # BOOTVERBOSE is off for the amd64 kernel, so we don't see the STARTING_INIT message
         # TODO: it would be nice if we had a message to detect userspace startup without requiring bootverbose
         bootverbose = False
-        boot_messages = [STARTING_INIT, "Hit \\[Enter\\] to boot ",
+        boot_messages = [STARTING_INIT, AUTOBOOT_PROMPT,
                          re.compile(r"Trying to mount root from .+\.\.\."),
-                         BOOT_FAILURE, BOOT_FAILURE2, BOOT_FAILURE3] + FATAL_ERROR_MESSAGES
+                         BOOT_FAILURE, BOOT_FAILURE2, BOOT_FAILURE3,
+                         NO_AUTOBOOT_PROMPT] + FATAL_ERROR_MESSAGES
         i = child.expect(boot_messages, timeout=5 * 60, timeout_msg="timeout before /sbin/init")
         # Skip 10s wait from x86 loader if we see the "Hit [Enter] to boot" message
         if i == 1:  # Hit Enter
+            if boot_alternate_kernel:
+                # Expected loader(8) to skip autoboot, fail
+                failure("expected autoboot disabled but got autoboot prompt")
             success("Got '", child.match.string, "' from loader")
             child.sendline("")
             i = child.expect(boot_messages, timeout=5 * 60, timeout_msg="timeout before /sbin/init")
+        if i == 6:  # loader(8) prompt
```

Comment:
There are so many messages now, I think we should consider using .index() like below to avoid future bugs.

Reply:
There is also another issue here, the alternate kernel is None if we are booting the default kernel, so I need to just send the "boot" command in that case.